### DISCUSSION TITLE
How do I create fill  edge region with white color and use as mask

### DISCUSSION THREAD

**Discussion by Webictbyleo** [ImageMagick/ImageMagick#4303] [Help]
I have gotten the edge using canny. The result is below
https://1drv.ms/u/s!AvpU2aoq2fBZhy1RCBPrjJ2AG-al?e=cE1Bkn

The original image is
https://1drv.ms/u/s!AvpU2aoq2fBZhy4aIzQ1Fs4KyAzV?e=Xzna0W

I want to fill the areas inside the edges with white color and then use it as mask to cut off the background on the original image. 

How do i achieve that??

**Comment by fmw42** [comment]
Your images are not visible to me. I am not sure we can access your OneDrive links.

**Reply by Webictbyleo** [reply]
https://1drv.ms/u/s!AvpU2aoq2fBZhy-065yRuDcz9RmM?e=lLsCvR

Above is the output of the edge detection on a black background.

**Comment by snibgo** [comment]
meshout.png has opaque white lines on a transparent black background.

> I want to fill the areas inside the edges with white color ...

But the white lines do not define an enclosed shape. No location can be chosen as a seed point for a flood fill.

**Reply by Webictbyleo** [reply]
The edges covers the area from the original image i want to enclose and fill with a white background. That way, i could use the shape as a mask. 

How do I go about it?

**Comment by Webictbyleo** [comment]
What I want to achieve is,
- Find edges of interest . I have done that and the result is up there. 
- Fill the area the edges covers with white color.  While the rest becomes black color
- Remove the black colored area leaving only the white colored area.
- Use the area covered by the white color as mask.

**Comment by Webictbyleo** [comment]
This is what I have done so for.
 First, I got the edges using the command below
 convert "bg.jpg" -canny 0x1+10%+30% -separate -evaluate-sequence max -equalize mask.png
 convert mask.png -transparent black maskalpha.png
 
 Above leaves me with only the edges. I want to fill the edge with white color and use it as mask on the main image.

**Comment by snibgo** [comment]
If we extract the alpha (so transparent becomes black and opaque becomes white), then flood-fill with red from the top-left corner:
```
magick meshout.png -alpha extract -fill Red -draw "color 0,0 floodfill" f.png
```
As you can see, flood-fill doesn't work, because all the transparent pixels on the left side are joined to all the others.
![f](https://user-images.githubusercontent.com/33812368/135676549-311b1adb-0c06-48ce-8f6d-3fff6c9fcfe4.png)
.

**Reply by Webictbyleo** [reply]
What i really want to do is just get the boundary area within the edges and then fill it with white. Not outside. 

If i know how to read the line geometry with image imagemagick, that will be a huge step.

**Comment by snibgo** [comment]
> ...  the boundary area within the edges ...

Which pixels are inside the "boundary", and which are outside? What rule determines the answer?

**Reply by Webictbyleo** [reply]
Look at the original image.

I want to select just the woman. I am able to mark the edges around her. So the boundary is area marked with white stroke.

**Comment by snibgo** [comment]
But you haven't drawn a white stroke around the woman. There is a gap at the top.

**Comment by fmw42** [comment]
Perhaps a better approach would be to work with color thresholding, morphology and connected-components to try to isolate the woman region.

**Comment by fmw42** [comment]
Your image is going to be hard by any image processing only method. But it can be done by AI/Deep Learning. See http://remove.bg.  Here is your example processed to remove the background. I am surprised that it actually handled the vertical line down the middle.

![remover-removebg-preview](https://user-images.githubusercontent.com/13458218/135686409-8a221804-9292-4d22-a6e9-f2042b29e106.png)
